src/device_manager.rs

```diff
@@ -11,6 +9,103 @@ use display::Display;
 use object::Object;
 use ffi;
 
+/// GdkDeviceManager — Functions for handling input devices
+///
+/// Description
+///
+/// In addition to a single pointer and keyboard for user interface input, GDK contains support
+/// for a variety of input devices, including graphics tablets, touchscreens and multiple
+/// pointers/keyboards interacting simultaneously with the user interface. Such input devices
+/// often have additional features, such as sub-pixel positioning information and additional
+/// device-dependent information.
+///
+/// In order to query the device hierarchy and be aware of changes in the device hierarchy
+/// (such as virtual devices being created or removed, or physical devices being plugged or
+/// unplugged), GDK provides DeviceManager.
+///
+/// By default, and if the platform supports it, GDK is aware of multiple keyboard/pointer pairs
+/// and multitouch devices. This behavior can be changed by calling
+/// DeviceManager::disable_multidevice() before Display::open(). There should rarely be a need to
+/// do that though, since GDK defaults to a compatibility mode in which it will emit just one
+/// enter/leave event pair for all devices on a window. To enable per-device enter/leave events
+/// and other multi-pointer interaction features, Window::set_support_multidevice() must be called
+/// on gdk::Windows (or gtk::Widget::set_support_multidevice() on widgets) window. See the
+/// Window::set_support_multidevice() documentation for more information.
+///
+/// On X11, multi-device support is implemented through XInput 2. Unless
+/// DeviceManager::disable_multidevice() is called, the XInput 2 GdkDeviceManager implementation
+/// will be used as the input source. Otherwise either the core or XInput 1 implementations will
+/// be used.
+///
+/// For simple applications that don’t have any special interest in input devices, the so-called
+/// "client pointer" provides a reasonable approximation to a simple setup with a single pointer
+/// and keyboard. The device that has been set as the client pointer can be accessed via
+/// DeviceManager::get_client_pointer().
+///
+/// Conceptually, in multidevice mode there are 2 device types. Virtual devices (or master devices)
+/// are represented by the pointer cursors and keyboard foci that are seen on the screen. Physical
+/// devices (or slave devices) represent the hardware that is controlling the virtual devices, and
+/// thus have no visible cursor on the screen.
+///
+/// Virtual devices are always paired, so there is a keyboard device for every pointer device.
+/// Associations between devices may be inspected through Device::get_associated_device().
+///
+/// There may be several virtual devices, and several physical devices could be controlling each of
+/// these virtual devices. Physical devices may also be “floating”, which means they are not attached
+/// to any virtual device.
+///
+/// #Master and slave devices
+///
+/// ``ìgnore
```

Comment:
Does this part render correctly? The syntax looks weird: two backticks never terminated?